datadog/resource_datadog_synthetics_test_.go

```diff
@@ -330,33 +376,77 @@ func buildSyntheticsTestStruct(d *schema.ResourceData) *datadogV1.SyntheticsTest
 	config := datadogV1.NewSyntheticsTestConfig([]datadogV1.SyntheticsAssertion{}, *request)
 	config.SetVariables([]datadogV1.SyntheticsBrowserVariable{})
 
+	// Deprecated path, the assertions field is replaced with assertion
 	if attr, ok := d.GetOk("assertions"); ok && attr != nil {
-		for _, attr := range attr.([]interface{}) {
-			assertion := datadogV1.SyntheticsAssertion{}
-			assertionMap := attr.(map[string]interface{})
+		for _, assertion := range attr.([]interface{}) {
+			assertionMap := assertion.(map[string]interface{})
 			if v, ok := assertionMap["type"]; ok {
 				assertionType := v.(string)
-				assertion.SetType(datadogV1.SyntheticsAssertionType(assertionType))
-			}
-			if v, ok := assertionMap["property"]; ok {
-				assertionProperty := v.(string)
-				assertion.SetProperty(assertionProperty)
-			}
-			if v, ok := assertionMap["operator"]; ok {
-				assertionOperator := v.(string)
-				assertion.SetOperator(datadogV1.SyntheticsAssertionOperator(assertionOperator))
+				if v, ok := assertionMap["operator"]; ok {
```

Comment:
It looks like this code path is pretty much the same as before, but some is further indented now. From the schema aren't `type` and `operator` siblings? Do the `operator` and `target` checks need to happen inside the `type` check?

Reply:
They are now required, so it's indented because we only create the structure when we already have the previous fields.